### PR TITLE
Switch Java code over to using Kotlin CRDTs

### PR DESCRIPTION
**Pull request by mmandlis**
copied from https://github.com/PolymerLabs/arcs/pull/4056

### CODE REVIEW

File: java/arcs/core/crdt/CrdtSet.kt [hunk 1]
```diff
@@ -21,17 +21,27 @@ import arcs.core.crdt.internal.Actor
...
 /** A [CrdtModel] capable of managing a set of items [T]. */
-class CrdtSet<T : Referencable> @JvmOverloads constructor(
+class CrdtSet<T : Referencable> (
     /** Initial data. */
     /* internal */ var _data: Data<T> = DataImpl(),
     /** Function to construct a new, empty [Data] object with a given [VersionMap]. */
-    private val dataBuilder: (VersionMap) -> Data<T> = { DataImpl(it) }
-) : CrdtModel<Data<T>, CrdtSet.IOperation<T>, Set<T>> {
+    private val dataBuilder: (VersionMap) -> Data<T>// = { DataImpl(it) }
+    ) : CrdtModel<Data<T>, CrdtSet.IOperation<T>, Set<T>> {
```
Comment (jasonwyatt) [inline java/arcs/core/crdt/CrdtSet.kt]: Indentation on this line is 4-spaces too much.

File: java/arcs/core/crdt/CrdtSet.kt [hunk 2]
```diff
@@ -21,17 +21,27 @@ import arcs.core.crdt.internal.Actor
...
+
+    // TODO(mmandlis): get rid of the secondary constructors, once Java code has migrated to Kotlin
+    // and there is no more need for @JvmOverloads annotation, that fails to compile to JS.
+    constructor() : this(DataImpl())
+
+    constructor(
+        data: Data<T>
+    ) : this(data, { DataImpl(it) })
```
Comment (jasonwyatt) [inline java/arcs/core/crdt/CrdtSet.kt]: A little cleaner would be:

```kotlin
constructor(data: Data<T>) : this(data) { DataImpl(it) }
```

File: java/arcs/core/crdt/CrdtSet.kt [hunk 3]
```diff
@@ -21,17 +21,27 @@ import arcs.core.crdt.internal.Actor
 import arcs.core.crdt.internal.VersionMap
 
 /** A [CrdtModel] capable of managing a set of items [T]. */
-class CrdtSet<T : Referencable>(
+class CrdtSet<T : Referencable> (
```
Comment (jasonwyatt) [inline java/arcs/core/crdt/CrdtSet.kt]: Nit: the space introduced here isn't needed.

File: javaharness/java/arcs/android/BUILD [hunk 4]
```diff
@@ -15,11 +15,12 @@ android_library(
     ],
     assets_dir = "",
     exports_manifest = 1,
-    idl_srcs = glob(["*.aidl"]),
+    # idl_srcs = glob(["*.aidl"]),
```
Comment (jasonwyatt) [inline javaharness/java/arcs/android/BUILD]: Rather than keeping this around, can it just be deleted?

File: java/arcs/core/crdt/CrdtSet.kt (hunk 1 above)
Comment (mmandlis) [reply java/arcs/core/crdt/CrdtSet.kt]: done

File: java/arcs/core/crdt/CrdtSet.kt (hunk 3 above)
Comment (mmandlis) [reply java/arcs/core/crdt/CrdtSet.kt]: removed

File: javaharness/java/arcs/android/BUILD (hunk 4 above)
Comment (mmandlis) [reply javaharness/java/arcs/android/BUILD]: removed

File: java/arcs/core/crdt/CrdtSet.kt (hunk 2 above)
Comment (mmandlis) [reply java/arcs/core/crdt/CrdtSet.kt]: absolutely agree, updated)

**Comment by mmandlis** [comment]
copybara shows as failed, but it was done manually and verified and approved.

**Pull request merged**